packages/mui-material/src/styles/overrides.d.ts

```diff
@@ -111,14 +112,25 @@ import { TooltipClassKey } from '../Tooltip';
 import { TouchRippleClassKey } from '../ButtonBase/TouchRipple';
 import { TypographyClassKey } from '../Typography';
 
-export type OverridesStyleRules<ClassKey extends string = string> = Record<
+export type OverridesStyleRules<
+  ClassKey extends string = string,
+  ComponentName = keyof ComponentsPropsList,
+  Theme = unknown,
+> = Record<
   ClassKey,
-  CSSInterpolation
+  | CSSInterpolation
+  | ((
+      props: (ComponentName extends keyof ComponentsPropsList
+        ? ComponentsPropsList[ComponentName]
+        : {}) & {
+        theme: Theme;
+      },
+    ) => CSSInterpolation)
 >;
 
-export type ComponentsOverrides = {
+export type ComponentsOverrides<Theme = unknown> = {
```

Comment:
add default type so it does not break the people who import `ComponentsOverrides` in what ever cases they are using.